packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -277,7 +285,7 @@ export default function MongoNav({
         admin={admin}
         hosts={hosts}
         currentProjectName={data?.currentProject?.projectName}
-        onLogout={onLogout}
+        onElementClick={onElementClick}
```

Comment:
So rather than tunnel this down to every level one thought I had was to make a local Context that just exposes this prop so any descendent component can access it. And then you can expose the function via a hook.

```jsx
// ProjectTrigger
const onElementClick = useOnElementClick();

// or even better. This works if there's no conditionals before it.
<div onClick={useOnElementClick()} />
```

Up to you.

Reply:
Also a more fun API for children is to allow this usage:

```jsx
<div name="logout" onClick={useOnElementClick(OnElementClick.Logout)} />
```

Such that `useOnElementClick` returns a function that takes in the event and then calls the `props.onElementClick` with the expectant API.

Reply:
👍